src/intellisense/langIntellisenseProvider.ts

```diff
@@ -628,6 +628,13 @@ class ReferenceCompletionProvider extends IntellisenseProvider implements vscode
         ['cite'],
     ];
 
+    protected readonly symbolProvider?:DocSymbolProvider;
```

Comment:
why use `protected`, and is `readonly` necessary?

Reply:
This is only for safty concerns since `symbolProvider` is not supposed to have any modification expect the initialization. If the `protected`, `readonly` introduce some problems, I can turn it into simple private.

Reply:
what if the safety concern?

Reply:
If any later modification will alter the symbolProvider in the `langIntellisenseProvider.ts` file from the maintenance perspective.

Reply:
I do not quilt get what you mean. The `?` to represent `undefined` seems to be a mistake, I'll remove it.

Reply:
Me neither, it seems like you are just copy the line above with `protected readonly`, which is obviously meaningless.

Reply:
Now is private and without `?`